src/c4/yml/emit.def.hpp

```diff
@@ -268,4 +276,148 @@ void Emitter<Writer>::_write_scalar(csubstr s)
 } // namespace yml
 } // namespace c4
 
+
+namespace c4 {
+namespace yml {
+
+template<class Writer>
+substr EmitterJSON<Writer>::emit(Tree const& t, size_t id, bool error_on_excess)
+{
+    this->_visit(t, id);
+    substr result = this->Writer::_get(error_on_excess);
+    return result;
+}
+
+template<class Writer>
+size_t EmitterJSON<Writer>::tell() const
+{
+    return this->Writer::m_pos;
+}
+
+template<class Writer>
+void EmitterJSON<Writer>::seek(size_t p)
+{
+    this->Writer::m_pos = p;
+}
+
+template<class Writer>
+size_t EmitterJSON<Writer>::_visit(Tree const& t, size_t id)
+{
+    if(t.is_stream(id))
+    {
+        ;
+    }
+    _do_visit(t, id);
+    if(t.is_stream(id))
+    {
+        _write("...\n");
+    }
+    return this->Writer::m_pos;
+}
+
+/** @todo this function is too complex. break it down into manageable pieces */
```

Comment:
this comment was for the original function. you can remove it here.

Reply:
done